### DISCUSSION TITLE
[3.x]: Sale priority is incorrect or confusing

### DISCUSSION THREAD

**Discussion by pixelmachine** [craftcms/commerce#3103] [General]
### What happened?

Perhaps I'm just not understanding but the sale priority does not seem to work as I would expect and the docs are pretty vague about it.

Here are some basic example scenarios (product cost is $100):
```
Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | false           | false
Sale 2 | Reduce by 30% | false           | false

Sale cost: $50 – as expected


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | false
Sale 2 | Reduce by 30% | false           | false

Sale cost: $80 – why is Sale 2 not applied?


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | true
Sale 2 | Reduce by 30% | false           | false

Sale cost: $80 – as expected


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | false
Sale 2 | Reduce by 30% | true            | false

Sale cost: $70 – as expected


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce to 90% | true (req'd)    | true
Sale 2 | Reduce by $20 | false           | false
Sale 3 | Reduce by 30% | false           | false

Sale cost: $90 – as expected


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce to 90% | true (req'd)    | false
Sale 2 | Reduce by $20 | false           | false
Sale 3 | Reduce by 30% | false           | false

Sale cost: $90 – why are Sale 2 & 3 not applied?
Expected: $40?


Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | false           | false
Sale 2 | Reduce to 90% | true (req'd)    | false
Sale 3 | Reduce by 30% | false           | false

Sale cost: $90 – why is Sale 3 not applied?
Expected: $60?
```

So there are two obvious issues I don't understand:
1. Setting 'Ignore previous...' seems to stop subsequent sales from applying
2. Setting any 'Reduce to' sale will disregard all others

I don't really have an issue with either but the UI does not make it clear what's going to happen at all.

### Craft CMS version

3.7.X

### Craft Commerce version

3.4.18

### PHP version

8.1

**Comment by lukeholder** [comment]
Hi @pixelmachine 

Can you please reproduce these sales and products in a fresh install of commerce 3.x and send a db dump with steps to reproduce to support@craftcms.com and we will reproduce and debug.

-Luke

**Comment by pixelmachine** [comment]
Ok, I will but I can't do it at the moment.

@pdaleramirez if you want to try it you can just setup 1 product and apply 3 sales to it as described above. Try changing the order and the 'Ignore ...' settings and see. There's not much else to it really.

**Comment by pdaleramirez** [comment]
@pixelmachine I've tested this and based on the mapping you've created all are expected behavior.

 When you enable the `Ignore previous matching sales if this sale matches.` on a sale it means that it will only apply that sale regardless of the order of the sales. The order of the sales affects the `Do not apply subsequent matching sales beyond applying this sale.` option.

**Comment by pixelmachine** [comment]
@pdaleramirez ok I understand that logic but in that case that wording doesn't make sense at all does it? It's not just previous sales it's "all other matching sales _unless_ a subsequent sale also has 'ignore previous'". So the order does also matter. For example:

```
Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | false
Sale 2 | Reduce by 30% | true            | false
```
In this case Sale 2 is applied, so...
> When you enable the Ignore previous matching sales if this sale matches. on a sale it means that it will only apply that sale regardless of the order of the sales

...is not really correct.

I don't really understand the reason for the logic either:
```
Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | false
Sale 2 | Reduce by 30% | false           | false
```
= Sale 1 applies but it's not clear why, especially when you could use 'ignore subsequent' to achieve the same logic:
```
Name   | Effect        | Ignore previous | Ignore subsequent
-------|---------------|-----------------|------------------    
Sale 1 | Reduce by $20 | true            | true
Sale 2 | Reduce by 30% | true            | false
```
= Sale 1 applies in an intentional way

**Comment by pixelmachine** [comment]
Also this should not be closed as the title is "Sale priority is incorrect or confusing" and the wording is confusing (or I would go as far as to say, wrong) as described in detail.

**Comment by pixelmachine** [comment]
What I have outlined is literally the definition of an Issue not a suggestion or discussion, why has it been banished here? @pdaleramirez
